### ISSUE TITLE
Add support for Lidl Auriol hg02832

### ISSUE THREAD

**Issue by OverSave** [merbanan/rtl_433#1161]
Hello, Please add the device Lidl Auriol hg02832.

`rtl_433 -r g001_433.93M_250k.cu8 -a -A
rtl_433 version 19.08-18-g8eecdbb branch master at 201909241811 inputs file rtl_tcp RTL-SDR
Use -h for usage help and see https://triq.org/ for documentation.
Trying conf file at "rtl_433.conf"...
Trying conf file at "/home/pi/.config/rtl_433/rtl_433.conf"...
Trying conf file at "/usr/local/etc/rtl_433/rtl_433.conf"...
Trying conf file at "/etc/rtl_433/rtl_433.conf"...
Consider using "-M newmodel" to transition to new model keys. This will become the default someday.
A table of changes and discussion is at https://github.com/merbanan/rtl_433/pull/986.
Registered 108 out of 138 device decoding protocols [ 1-4 8 11-12 15-17 19-21 23 25-26 29-36 38-60 63 67-71 73-100 102-103 108-116 119 121 124-128 131-138 ]
Test mode active. Reading samples from file: g001_433.93M_250k.cu8
Detected OOK package    @0.211772s
Analyzing pulses...
Total count:   44,  width: 40.22 ms             (10056 S)
Pulse width distribution:
 [ 0] count:   21,  width:  252 us [244;308]    (  63 S)
 [ 1] count:    3,  width:  860 us [852;864]    ( 215 S)
 [ 2] count:   20,  width:  616 us [608;620]    ( 154 S)
Gap width distribution:
 [ 0] count:    4,  width:  848 us [840;856]    ( 212 S)
 [ 1] count:   20,  width:  600 us [600;608]    ( 150 S)
 [ 2] count:   19,  width:  236 us [236;248]    (  59 S)
Pulse period distribution:
 [ 0] count:    1,  width: 1148 us [1148;1148]  ( 287 S)
 [ 1] count:    3,  width: 1708 us [1700;1720]  ( 427 S)
 [ 2] count:   39,  width:  852 us [848;860]    ( 213 S)
Level estimates [high, low]:   1676,     17
RSSI: -9.9 dB SNR: 19.7 dB Noise: -29.6 dB
Frequency offsets [F1, F2]:     566,      0     (+2.2 kHz, +0.0 kHz)
Guessing modulation: Pulse Width Modulation with sync/delimiter
Attempting demodulation... short_width: 252, long_width: 616, reset_limit: 860, sync_width: 860
Use a flex decoder with -X 'n=name,m=OOK_PWM,s=252,l=616,r=860,g=0,t=0,y=860'
pulse_demod_pwm(): Analyzer Device
bitbuffer:: Number of rows: 2
[00] { 1} 80             : 1
[01] {40} dd ac 5f 92 10 : 11011101 10101100 01011111 10010010 00010000

Detected OOK package    @0.312884s
Analyzing pulses...
Total count:   44,  width: 40.24 ms             (10061 S)
Pulse width distribution:
 [ 0] count:    1,  width:  328 us [328;328]    (  82 S)
 [ 1] count:    3,  width:  860 us [860;860]    ( 215 S)
 [ 2] count:   20,  width:  248 us [244;260]    (  62 S)
 [ 3] count:   20,  width:  616 us [608;624]    ( 154 S)
Gap width distribution:
 [ 0] count:    4,  width:  848 us [844;856]    ( 212 S)
 [ 1] count:   20,  width:  600 us [596;608]    ( 150 S)
 [ 2] count:   19,  width:  236 us [232;248]    (  59 S)
Pulse period distribution:
 [ 0] count:    1,  width: 1176 us [1176;1176]  ( 294 S)
 [ 1] count:    3,  width: 1708 us [1704;1716]  ( 427 S)
 [ 2] count:   39,  width:  852 us [844;864]    ( 213 S)
Level estimates [high, low]:   1756,     16
RSSI: -9.7 dB SNR: 20.1 dB Noise: -29.8 dB
Frequency offsets [F1, F2]:     651,      0     (+2.5 kHz, +0.0 kHz)
Guessing modulation: No clue...

Detected OOK package    @0.414024s
Analyzing pulses...
Total count:   44,  width: 40.24 ms             (10061 S)
Pulse width distribution:
 [ 0] count:    1,  width:  332 us [332;332]    (  83 S)
 [ 1] count:    3,  width:  860 us [856;864]    ( 215 S)
 [ 2] count:   20,  width:  248 us [240;260]    (  62 S)
 [ 3] count:   20,  width:  616 us [608;624]    ( 154 S)
Gap width distribution:
 [ 0] count:    4,  width:  844 us [840;848]    ( 211 S)
 [ 1] count:   20,  width:  600 us [596;612]    ( 150 S)
 [ 2] count:   19,  width:  236 us [232;244]    (  59 S)
Pulse period distribution:
 [ 0] count:    1,  width: 1172 us [1172;1172]  ( 293 S)
 [ 1] count:    3,  width: 1708 us [1704;1712]  ( 427 S)
 [ 2] count:   39,  width:  852 us [844;864]    ( 213 S)
Level estimates [high, low]:   1681,     17
RSSI: -9.9 dB SNR: 19.7 dB Noise: -29.6 dB
Frequency offsets [F1, F2]:     405,      0     (+1.5 kHz, +0.0 kHz)
Guessing modulation: No clue...

Detected OOK package    @0.515160s
Analyzing pulses...
Total count:   44,  width: 40.24 ms             (10061 S)
Pulse width distribution:
 [ 0] count:   21,  width:  252 us [244;328]    (  63 S)
 [ 1] count:    3,  width:  860 us [860;864]    ( 215 S)
 [ 2] count:   20,  width:  616 us [608;624]    ( 154 S)
Gap width distribution:
 [ 0] count:    4,  width:  844 us [844;852]    ( 211 S)
 [ 1] count:   20,  width:  600 us [596;616]    ( 150 S)
 [ 2] count:   19,  width:  236 us [228;248]    (  59 S)
Pulse period distribution:
 [ 0] count:    1,  width: 1172 us [1172;1172]  ( 293 S)
 [ 1] count:    3,  width: 1708 us [1708;1716]  ( 427 S)
 [ 2] count:   39,  width:  852 us [844;868]    ( 213 S)
Level estimates [high, low]:   1736,     19
RSSI: -9.7 dB SNR: 19.4 dB Noise: -29.1 dB
Frequency offsets [F1, F2]:   -1036,      0     (-4.0 kHz, +0.0 kHz)
Guessing modulation: Pulse Width Modulation with sync/delimiter
Attempting demodulation... short_width: 252, long_width: 616, reset_limit: 856, sync_width: 860
Use a flex decoder with -X 'n=name,m=OOK_PWM,s=252,l=616,r=856,g=0,t=0,y=860'
pulse_demod_pwm(): Analyzer Device
bitbuffer:: Number of rows: 2
[00] { 1} 80             : 1
[01] {40} dd ac 5f 92 10 : 11011101 10101100 01011111 10010010 00010000

Detected OOK package    @0.616296s
Analyzing pulses...
Total count:   44,  width: 40.24 ms             (10061 S)
Pulse width distribution:
 [ 0] count:    1,  width:  332 us [332;332]    (  83 S)
 [ 1] count:    3,  width:  860 us [860;864]    ( 215 S)
 [ 2] count:   20,  width:  248 us [244;256]    (  62 S)
 [ 3] count:   20,  width:  616 us [608;624]    ( 154 S)
Gap width distribution:
 [ 0] count:    4,  width:  844 us [840;852]    ( 211 S)
 [ 1] count:   20,  width:  600 us [600;608]    ( 150 S)
 [ 2] count:   19,  width:  236 us [232;248]    (  59 S)
Pulse period distribution:
 [ 0] count:    1,  width: 1172 us [1172;1172]  ( 293 S)
 [ 1] count:    3,  width: 1708 us [1704;1712]  ( 427 S)
 [ 2] count:   39,  width:  852 us [844;868]    ( 213 S)
Level estimates [high, low]:   1726,     11
RSSI: -9.8 dB SNR: 21.6 dB Noise: -31.4 dB
Frequency offsets [F1, F2]:     467,      0     (+1.8 kHz, +0.0 kHz)
Guessing modulation: No clue...

Detected OOK package    @0.717436s
Analyzing pulses...
Total count:   44,  width: 40.24 ms             (10060 S)
Pulse width distribution:
 [ 0] count:    1,  width:  324 us [324;324]    (  81 S)
 [ 1] count:    3,  width:  860 us [856;864]    ( 215 S)
 [ 2] count:   20,  width:  248 us [244;256]    (  62 S)
 [ 3] count:   20,  width:  616 us [608;620]    ( 154 S)
Gap width distribution:
 [ 0] count:    4,  width:  848 us [844;852]    ( 212 S)
 [ 1] count:   20,  width:  600 us [600;608]    ( 150 S)
 [ 2] count:   19,  width:  236 us [232;248]    (  59 S)
Pulse period distribution:
 [ 0] count:    1,  width: 1172 us [1172;1172]  ( 293 S)
 [ 1] count:    3,  width: 1708 us [1704;1712]  ( 427 S)
 [ 2] count:   39,  width:  852 us [844;860]    ( 213 S)
Level estimates [high, low]:   1723,     14
RSSI: -9.8 dB SNR: 20.6 dB Noise: -30.4 dB
Frequency offsets [F1, F2]:      53,      0     (+0.2 kHz, +0.0 kHz)
Guessing modulation: No clue...

Detected OOK package    @0.818576s
Analyzing pulses...
Total count:   44,  width: 40.24 ms             (10060 S)
Pulse width distribution:
 [ 0] count:    1,  width:  324 us [324;324]    (  81 S)
 [ 1] count:    3,  width:  860 us [856;864]    ( 215 S)
 [ 2] count:   20,  width:  248 us [244;256]    (  62 S)
 [ 3] count:   20,  width:  616 us [612;624]    ( 154 S)
Gap width distribution:
 [ 0] count:    4,  width:  848 us [840;852]    ( 212 S)
 [ 1] count:   20,  width:  600 us [596;608]    ( 150 S)
 [ 2] count:   19,  width:  236 us [232;244]    (  59 S)
Pulse period distribution:
 [ 0] count:    1,  width: 1164 us [1164;1164]  ( 291 S)
 [ 1] count:    3,  width: 1708 us [1708;1716]  ( 427 S)
 [ 2] count:   39,  width:  852 us [848;864]    ( 213 S)
Level estimates [high, low]:   1752,     16
RSSI: -9.7 dB SNR: 20.1 dB Noise: -29.8 dB
Frequency offsets [F1, F2]:     777,      0     (+3.0 kHz, +0.0 kHz)
Guessing modulation: No clue...

Detected OOK package    @0.919712s
Analyzing pulses...
Total count:   44,  width: 40.24 ms             (10059 S)
Pulse width distribution:
 [ 0] count:    1,  width:  328 us [328;328]    (  82 S)
 [ 1] count:    3,  width:  864 us [864;868]    ( 216 S)
 [ 2] count:   20,  width:  248 us [244;256]    (  62 S)
 [ 3] count:   20,  width:  616 us [604;628]    ( 154 S)
Gap width distribution:
 [ 0] count:    4,  width:  840 us [836;844]    ( 210 S)
 [ 1] count:   20,  width:  600 us [596;608]    ( 150 S)
 [ 2] count:   19,  width:  236 us [232;248]    (  59 S)
Pulse period distribution:
 [ 0] count:    1,  width: 1164 us [1164;1164]  ( 291 S)
 [ 1] count:    3,  width: 1708 us [1708;1712]  ( 427 S)
 [ 2] count:   39,  width:  852 us [844;868]    ( 213 S)
Level estimates [high, low]:   1711,      7
RSSI: -9.8 dB SNR: 23.3 dB Noise: -33.1 dB
Frequency offsets [F1, F2]:     395,      0     (+1.5 kHz, +0.0 kHz)
Guessing modulation: No clue...

Detected OOK package    @1.020852s
Analyzing pulses...
Total count:   44,  width: 40.24 ms             (10060 S)
Pulse width distribution:
 [ 0] count:   21,  width:  252 us [244;320]    (  63 S)
 [ 1] count:    3,  width:  860 us [856;868]    ( 215 S)
 [ 2] count:   20,  width:  616 us [612;628]    ( 154 S)
Gap width distribution:
 [ 0] count:    4,  width:  844 us [840;848]    ( 211 S)
 [ 1] count:   20,  width:  600 us [600;612]    ( 150 S)
 [ 2] count:   19,  width:  236 us [228;244]    (  59 S)
Pulse period distribution:
 [ 0] count:    1,  width: 1160 us [1160;1160]  ( 290 S)
 [ 1] count:    3,  width: 1708 us [1704;1716]  ( 427 S)
 [ 2] count:   39,  width:  852 us [844;864]    ( 213 S)
Level estimates [high, low]:   1752,     18
RSSI: -9.7 dB SNR: 19.6 dB Noise: -29.4 dB
Frequency offsets [F1, F2]:    -993,      0     (-3.8 kHz, +0.0 kHz)
Guessing modulation: Pulse Width Modulation with sync/delimiter
Attempting demodulation... short_width: 252, long_width: 616, reset_limit: 852, sync_width: 860
Use a flex decoder with -X 'n=name,m=OOK_PWM,s=252,l=616,r=852,g=0,t=0,y=860'
pulse_demod_pwm(): Analyzer Device
bitbuffer:: Number of rows: 2
[00] { 1} 80             : 1
[01] {40} dd ac 5f 92 10 : 11011101 10101100 01011111 10010010 00010000

Detected OOK package    @1.121984s
Analyzing pulses...
Total count:   44,  width: 40.24 ms             (10060 S)
Pulse width distribution:
 [ 0] count:   21,  width:  252 us [244;324]    (  63 S)
 [ 1] count:    3,  width:  860 us [860;864]    ( 215 S)
 [ 2] count:   20,  width:  616 us [604;624]    ( 154 S)
Gap width distribution:
 [ 0] count:    4,  width:  844 us [844;852]    ( 211 S)
 [ 1] count:   20,  width:  600 us [596;608]    ( 150 S)
 [ 2] count:   19,  width:  236 us [232;248]    (  59 S)
Pulse period distribution:
 [ 0] count:    1,  width: 1168 us [1168;1168]  ( 292 S)
 [ 1] count:    3,  width: 1708 us [1708;1712]  ( 427 S)
 [ 2] count:   39,  width:  852 us [844;864]    ( 213 S)
Level estimates [high, low]:   1682,     14
RSSI: -9.9 dB SNR: 20.5 dB Noise: -30.4 dB
Frequency offsets [F1, F2]:     168,      0     (+0.6 kHz, +0.0 kHz)
Guessing modulation: Pulse Width Modulation with sync/delimiter
Attempting demodulation... short_width: 252, long_width: 616, reset_limit: 856, sync_width: 860
Use a flex decoder with -X 'n=name,m=OOK_PWM,s=252,l=616,r=856,g=0,t=0,y=860'
pulse_demod_pwm(): Analyzer Device
bitbuffer:: Number of rows: 2
[00] { 1} 80             : 1
[01] {40} dd ac 5f 92 10 : 11011101 10101100 01011111 10010010 00010000

Detected OOK package    @1.333292s
Analyzing pulses...
Total count:   44,  width: 40.24 ms             (10061 S)
Pulse width distribution:
 [ 0] count:    1,  width:  324 us [324;324]    (  81 S)
 [ 1] count:    3,  width:  860 us [856;864]    ( 215 S)
 [ 2] count:   20,  width:  248 us [244;256]    (  62 S)
 [ 3] count:   20,  width:  616 us [608;624]    ( 154 S)
Gap width distribution:
 [ 0] count:    4,  width:  844 us [844;852]    ( 211 S)
 [ 1] count:   20,  width:  600 us [596;612]    ( 150 S)
 [ 2] count:   19,  width:  236 us [232;244]    (  59 S)
Pulse period distribution:
 [ 0] count:    1,  width: 1168 us [1168;1168]  ( 292 S)
 [ 1] count:    3,  width: 1708 us [1704;1712]  ( 427 S)
 [ 2] count:   39,  width:  852 us [844;864]    ( 213 S)
Level estimates [high, low]:   1739,     12
RSSI: -9.7 dB SNR: 21.3 dB Noise: -31.0 dB
Frequency offsets [F1, F2]:     -30,      0     (-0.1 kHz, +0.0 kHz)
Guessing modulation: No clue...

Detected OOK package    @1.434428s
Analyzing pulses...
Total count:   44,  width: 40.24 ms             (10060 S)
Pulse width distribution:
 [ 0] count:    1,  width:  324 us [324;324]    (  81 S)
 [ 1] count:    3,  width:  860 us [860;864]    ( 215 S)
 [ 2] count:   20,  width:  248 us [244;256]    (  62 S)
 [ 3] count:   20,  width:  616 us [608;620]    ( 154 S)
Gap width distribution:
 [ 0] count:    4,  width:  844 us [844;848]    ( 211 S)
 [ 1] count:   20,  width:  600 us [596;608]    ( 150 S)
 [ 2] count:   19,  width:  236 us [232;244]    (  59 S)
Pulse period distribution:
 [ 0] count:    1,  width: 1168 us [1168;1168]  ( 292 S)
 [ 1] count:    3,  width: 1708 us [1708;1712]  ( 427 S)
 [ 2] count:   39,  width:  852 us [844;864]    ( 213 S)
Level estimates [high, low]:   1749,     18
RSSI: -9.7 dB SNR: 19.6 dB Noise: -29.4 dB
Frequency offsets [F1, F2]:   -1158,      0     (-4.4 kHz, +0.0 kHz)
Guessing modulation: No clue...

Detected OOK package    @1.535572s
Analyzing pulses...
Total count:   44,  width: 40.23 ms             (10058 S)
Pulse width distribution:
 [ 0] count:   21,  width:  252 us [248;316]    (  63 S)
 [ 1] count:    3,  width:  860 us [856;864]    ( 215 S)
 [ 2] count:   20,  width:  616 us [608;624]    ( 154 S)
Gap width distribution:
 [ 0] count:    4,  width:  848 us [844;852]    ( 212 S)
 [ 1] count:   20,  width:  600 us [596;612]    ( 150 S)
 [ 2] count:   19,  width:  236 us [232;244]    (  59 S)
Pulse period distribution:
 [ 0] count:    1,  width: 1160 us [1160;1160]  ( 290 S)
 [ 1] count:    3,  width: 1708 us [1704;1716]  ( 427 S)
 [ 2] count:   39,  width:  852 us [844;864]    ( 213 S)
Level estimates [high, low]:   1769,     15
RSSI: -9.7 dB SNR: 20.4 dB Noise: -30.1 dB
Frequency offsets [F1, F2]:     282,      0     (+1.1 kHz, +0.0 kHz)
Guessing modulation: Pulse Width Modulation with sync/delimiter
Attempting demodulation... short_width: 252, long_width: 616, reset_limit: 856, sync_width: 860
Use a flex decoder with -X 'n=name,m=OOK_PWM,s=252,l=616,r=856,g=0,t=0,y=860'
pulse_demod_pwm(): Analyzer Device
bitbuffer:: Number of rows: 2
[00] { 1} 80             : 1
[01] {40} dd ac 5f 92 10 : 11011101 10101100 01011111 10010010 00010000

Detected OOK package    @1.636708s
Analyzing pulses...
Total count:   44,  width: 40.23 ms             (10058 S)
Pulse width distribution:
 [ 0] count:   21,  width:  252 us [244;320]    (  63 S)
 [ 1] count:    3,  width:  860 us [860;864]    ( 215 S)
 [ 2] count:   20,  width:  616 us [604;624]    ( 154 S)
Gap width distribution:
 [ 0] count:    4,  width:  844 us [836;852]    ( 211 S)
 [ 1] count:   20,  width:  600 us [596;608]    ( 150 S)
 [ 2] count:   19,  width:  236 us [232;248]    (  59 S)
Pulse period distribution:
 [ 0] count:    1,  width: 1156 us [1156;1156]  ( 289 S)
 [ 1] count:    3,  width: 1708 us [1708;1716]  ( 427 S)
 [ 2] count:   39,  width:  852 us [844;868]    ( 213 S)
Level estimates [high, low]:   1751,     10
RSSI: -9.7 dB SNR: 22.0 dB Noise: -31.7 dB
Frequency offsets [F1, F2]:     603,      0     (+2.3 kHz, +0.0 kHz)
Guessing modulation: Pulse Width Modulation with sync/delimiter
Attempting demodulation... short_width: 252, long_width: 616, reset_limit: 856, sync_width: 860
Use a flex decoder with -X 'n=name,m=OOK_PWM,s=252,l=616,r=856,g=0,t=0,y=860'
pulse_demod_pwm(): Analyzer Device
bitbuffer:: Number of rows: 2
[00] { 1} 80             : 1
[01] {40} dd ac 5f 92 10 : 11011101 10101100 01011111 10010010 00010000

Detected OOK package    @1.737844s
Analyzing pulses...
Total count:   44,  width: 40.24 ms             (10059 S)
Pulse width distribution:
 [ 0] count:   21,  width:  252 us [244;316]    (  63 S)
 [ 1] count:    3,  width:  860 us [860;864]    ( 215 S)
 [ 2] count:   20,  width:  616 us [608;624]    ( 154 S)
Gap width distribution:
 [ 0] count:    4,  width:  848 us [844;856]    ( 212 S)
 [ 1] count:   20,  width:  604 us [600;612]    ( 151 S)
 [ 2] count:   19,  width:  236 us [228;244]    (  59 S)
Pulse period distribution:
 [ 0] count:    1,  width: 1160 us [1160;1160]  ( 290 S)
 [ 1] count:    3,  width: 1708 us [1708;1716]  ( 427 S)
 [ 2] count:   39,  width:  852 us [844;864]    ( 213 S)
Level estimates [high, low]:   1742,     13
RSSI: -9.7 dB SNR: 20.9 dB Noise: -30.7 dB
Frequency offsets [F1, F2]:    -190,      0     (-0.7 kHz, +0.0 kHz)
Guessing modulation: Pulse Width Modulation with sync/delimiter
Attempting demodulation... short_width: 252, long_width: 616, reset_limit: 860, sync_width: 860
Use a flex decoder with -X 'n=name,m=OOK_PWM,s=252,l=616,r=860,g=0,t=0,y=860'
pulse_demod_pwm(): Analyzer Device
bitbuffer:: Number of rows: 2
[00] { 1} 80             : 1
[01] {40} dd ac 5f 92 10 : 11011101 10101100 01011111 10010010 00010000

Detected OOK package    @1.838984s
Analyzing pulses...
Total count:   44,  width: 40.24 ms             (10059 S)
Pulse width distribution:
 [ 0] count:   21,  width:  252 us [248;312]    (  63 S)
 [ 1] count:    3,  width:  856 us [852;864]    ( 214 S)
 [ 2] count:   20,  width:  616 us [612;624]    ( 154 S)
Gap width distribution:
 [ 0] count:    4,  width:  848 us [844;856]    ( 212 S)
 [ 1] count:   20,  width:  600 us [600;604]    ( 150 S)
 [ 2] count:   19,  width:  236 us [232;244]    (  59 S)
Pulse period distribution:
 [ 0] count:    1,  width: 1156 us [1156;1156]  ( 289 S)
 [ 1] count:    3,  width: 1708 us [1700;1716]  ( 427 S)
 [ 2] count:   39,  width:  852 us [848;860]    ( 213 S)
Level estimates [high, low]:   1682,     16
RSSI: -9.9 dB SNR: 20.0 dB Noise: -29.8 dB
Frequency offsets [F1, F2]:     445,      0     (+1.7 kHz, +0.0 kHz)
Guessing modulation: Pulse Width Modulation with sync/delimiter
Attempting demodulation... short_width: 252, long_width: 616, reset_limit: 860, sync_width: 856
Use a flex decoder with -X 'n=name,m=OOK_PWM,s=252,l=616,r=860,g=0,t=0,y=856'
pulse_demod_pwm(): Analyzer Device
bitbuffer:: Number of rows: 2
[00] { 1} 80             : 1
[01] {40} dd ac 5f 92 10 : 11011101 10101100 01011111 10010010 00010000

Detected OOK package    @1.940120s
Analyzing pulses...
Total count:   44,  width: 40.23 ms             (10058 S)
Pulse width distribution:
 [ 0] count:   21,  width:  252 us [244;316]    (  63 S)
 [ 1] count:    3,  width:  864 us [864;868]    ( 216 S)
 [ 2] count:   20,  width:  612 us [604;620]    ( 153 S)
Gap width distribution:
 [ 0] count:    4,  width:  840 us [836;848]    ( 210 S)
 [ 1] count:   20,  width:  600 us [596;608]    ( 150 S)
 [ 2] count:   19,  width:  236 us [232;248]    (  59 S)
Pulse period distribution:
 [ 0] count:    1,  width: 1152 us [1152;1152]  ( 288 S)
 [ 1] count:    3,  width: 1708 us [1708;1716]  ( 427 S)
 [ 2] count:   39,  width:  852 us [848;864]    ( 213 S)
Level estimates [high, low]:   1674,     18
RSSI: -9.9 dB SNR: 19.5 dB Noise: -29.4 dB
Frequency offsets [F1, F2]:     673,      0     (+2.6 kHz, +0.0 kHz)
Guessing modulation: Pulse Width Modulation with sync/delimiter
Attempting demodulation... short_width: 252, long_width: 612, reset_limit: 852, sync_width: 864
Use a flex decoder with -X 'n=name,m=OOK_PWM,s=252,l=612,r=852,g=0,t=0,y=864'
pulse_demod_pwm(): Analyzer Device
bitbuffer:: Number of rows: 2
[00] { 1} 80             : 1
[01] {40} dd ac 5f 92 10 : 11011101 10101100 01011111 10010010 00010000

Detected OOK package    @2.041256s
Analyzing pulses...
Total count:   44,  width: 40.23 ms             (10058 S)
Pulse width distribution:
 [ 0] count:   21,  width:  252 us [244;316]    (  63 S)
 [ 1] count:    3,  width:  860 us [860;864]    ( 215 S)
 [ 2] count:   20,  width:  616 us [608;620]    ( 154 S)
Gap width distribution:
 [ 0] count:    4,  width:  844 us [844;852]    ( 211 S)
 [ 1] count:   20,  width:  604 us [600;612]    ( 151 S)
 [ 2] count:   19,  width:  236 us [236;248]    (  59 S)
Pulse period distribution:
 [ 0] count:    1,  width: 1160 us [1160;1160]  ( 290 S)
 [ 1] count:    3,  width: 1708 us [1708;1712]  ( 427 S)
 [ 2] count:   39,  width:  852 us [848;860]    ( 213 S)
Level estimates [high, low]:   1686,     11
RSSI: -9.9 dB SNR: 21.5 dB Noise: -31.4 dB
Frequency offsets [F1, F2]:     246,      0     (+0.9 kHz, +0.0 kHz)
Guessing modulation: Pulse Width Modulation with sync/delimiter
Attempting demodulation... short_width: 252, long_width: 616, reset_limit: 856, sync_width: 860
Use a flex decoder with -X 'n=name,m=OOK_PWM,s=252,l=616,r=856,g=0,t=0,y=860'
pulse_demod_pwm(): Analyzer Device
bitbuffer:: Number of rows: 2
[00] { 1} 80             : 1
[01] {40} dd ac 5f 92 10 : 11011101 10101100 01011111 10010010 00010000

Detected OOK package    @2.142392s
Analyzing pulses...
Total count:   44,  width: 40.23 ms             (10058 S)
Pulse width distribution:
 [ 0] count:   21,  width:  252 us [244;316]    (  63 S)
 [ 1] count:    3,  width:  856 us [856;860]    ( 214 S)
 [ 2] count:   20,  width:  612 us [608;624]    ( 153 S)
Gap width distribution:
 [ 0] count:    4,  width:  848 us [840;856]    ( 212 S)
 [ 1] count:   20,  width:  604 us [600;612]    ( 151 S)
 [ 2] count:   19,  width:  236 us [236;248]    (  59 S)
Pulse period distribution:
 [ 0] count:    1,  width: 1156 us [1156;1156]  ( 289 S)
 [ 1] count:    3,  width: 1708 us [1704;1716]  ( 427 S)
 [ 2] count:   39,  width:  852 us [848;864]    ( 213 S)
Level estimates [high, low]:   1678,     20
RSSI: -9.9 dB SNR: 19.0 dB Noise: -28.9 dB
Frequency offsets [F1, F2]:   -1102,      0     (-4.2 kHz, +0.0 kHz)
Guessing modulation: Pulse Width Modulation with sync/delimiter
Attempting demodulation... short_width: 252, long_width: 612, reset_limit: 860, sync_width: 856
Use a flex decoder with -X 'n=name,m=OOK_PWM,s=252,l=612,r=860,g=0,t=0,y=856'
pulse_demod_pwm(): Analyzer Device
bitbuffer:: Number of rows: 2
[00] { 1} 80             : 1
[01] {40} dd ac 5f 92 10 : 11011101 10101100 01011111 10010010 00010000

Detected OOK package    @2.243532s
Analyzing pulses...
Total count:   44,  width: 40.22 ms             (10056 S)
Pulse width distribution:
 [ 0] count:   21,  width:  252 us [244;316]    (  63 S)
 [ 1] count:    3,  width:  860 us [856;868]    ( 215 S)
 [ 2] count:   20,  width:  612 us [608;620]    ( 153 S)
Gap width distribution:
 [ 0] count:    4,  width:  844 us [836;852]    ( 211 S)
 [ 1] count:   20,  width:  600 us [600;612]    ( 150 S)
 [ 2] count:   19,  width:  236 us [236;244]    (  59 S)
Pulse period distribution:
 [ 0] count:    1,  width: 1152 us [1152;1152]  ( 288 S)
 [ 1] count:    3,  width: 1708 us [1704;1720]  ( 427 S)
 [ 2] count:   39,  width:  852 us [844;864]    ( 213 S)
Level estimates [high, low]:   1712,     12
RSSI: -9.8 dB SNR: 21.2 dB Noise: -31.0 dB
Frequency offsets [F1, F2]:    -770,      0     (-2.9 kHz, +0.0 kHz)
Guessing modulation: Pulse Width Modulation with sync/delimiter
Attempting demodulation... short_width: 252, long_width: 612, reset_limit: 856, sync_width: 860
Use a flex decoder with -X 'n=name,m=OOK_PWM,s=252,l=612,r=856,g=0,t=0,y=860'
pulse_demod_pwm(): Analyzer Device
bitbuffer:: Number of rows: 2
[00] { 1} 80             : 1
[01] {40} dd ac 5f 92 10 : 11011101 10101100 01011111 10010010 00010000
`

![auriol hg02832](https://user-images.githubusercontent.com/49719788/66301989-70972d80-e8f8-11e9-8f75-d8050f9dcb73.jpg)

**Comment by zuckschwerdt** [comment]
The bits before temperature are likely high bits for temperature. 7 bit isn't enough. common is 12 bit. so 4 bit status (batt...) and 12 bit temp. the last byte is a checksum. seems easy enough, I'll add a decoder once we have samples: https://github.com/merbanan/rtl_433#supporting-additional-devices-and-test-data

**Comment by OverSave** [comment]
i added a fork and pull request

**Comment by zuckschwerdt** [comment]
I'll add a decoder soon. In the meantime:
```
rtl_433 -X 'n=auriol-hg02832,
  m=OOK_PWM,
  s=252,
  l=612,
  y=860,
  g=750,
  r=62990,
  invert,
  bits>=40,
  get=@0:{8}:id,
  get=@8:{8}:humi,
  get=@16:{4}:flags,
  get=@20:{12}:temp,
  get=@32:{8}:chk' tests/auriol/06/*.cu8
```

**Comment by zuckschwerdt** [comment]
To crack the checksum, can you collect a range of unique codes? Just a text file with lines like this:
```
{40}2254009d06
```

**Comment by merbanan** [comment]
2254009d06
225300b308
225210b68f
225c10b88f
225c20ba28
225f20c30b
225f00c41a
225f10c1ac
225f20bebf

225210b68f -> 2^6=4
225c10b88f -> c^8=4

Looks linear.

**Comment by merbanan** [comment]
@OverSave we need a large amount of messages to be able to figure out the checksum. I think that the id changes when you change the batteries. So run the command for a day, change the battery and repeat. Then post all the messages you get.

**Comment by OverSave** [comment]
this command?

rtl_433 -X 'n=auriol-hg02832, m=OOK_PWM, s=252, l=612, y=860, g=750, r=62990, invert, bits>=40, get=@0:{8}:id, get=@8:{8}:humi, get=@16:{4}:flags, get=@20:{12}:temp, get=@32:{8}:chk' tests/auriol/06/*.cu8

**Comment by zuckschwerdt** [comment]
Yes, just not the `tests/auriol/06/*.cu8` part. That's offline reading, you want live reception. Also with `-F json:yourlog.txt` to collect the output. And you can remove all the `,get=...` parameters if you don't care about the debug output. Makes the log shorter. Use a different log name (date suffix?) each time to keep track.

**Comment by merbanan** [comment]
Just run with rtl_433 -X 'n=auriol-hg02832, m=OOK_PWM, s=252, l=612, y=860, g=750, r=62990, invert, bits>=40, get=@0:{8}:id, get=@8:{8}:humi, get=@16:{4}:flags, get=@20:{12}:temp, get=@32:{8}:chk'

You can redirect the decoded messages to a file.

**Comment by OverSave** [comment]
I wanted to create a 24 hour log but after 7 hours it was canceled. I hope the 7 hours are enough

**Comment by zuckschwerdt** [comment]
Thanks! Add here as a zip would be ok too. I'll filter the unique codes and see what we get.

**Comment by zuckschwerdt** [comment]
Strangely I don't see how the keys are constructed. More unique codes will be helpful. Can you reset the ID (remove and replace the battery) and record more codes? Just drop a zip here.

**Comment by OverSave** [comment]
all right

**Comment by OverSave** [comment]
okay here are another 20 hours.

[20191011_newbat_log.zip](https://github.com/merbanan/rtl_433/files/3720369/20191011_newbat_log.zip)

**Comment by zuckschwerdt** [comment]
The ID didn't change. But here is a try anyway:
In addition to the keys above we got two double bit changes to work with:
- bit  9+10 -> mask 0xbb
- bit 16+17 -> mask 0x47

If we reasonably assume key 10 is the same as key 26, then key 9 must be 0x86^0xbb=0x3d.
If we then expand the observed pattern (poly 0x31) one bit left/up we suspect key 8 = 0x7a.
When we xor key 8 and 9 we get 0x47. This fits The double bit change at 16+17.

In summary. They tried to use a CRC-8 poly 0x31, but shifted the wrong way and also (accidentally?) reset the key every new byte.
The algo now is: CRC-8 poly 0x98, init 0x7a, for each byte then xor all.
(key stream is 7a 3d 86 43 b9 c4 62 31 repeated 4 times.)

**Comment by OverSave** [comment]
someone has already worked on that, does that help?

https://github.com/alex-konshin/f007th-rpi/blob/master/SensorsData.hpp

**Comment by zuckschwerdt** [comment]
It's all figured out already, decoder coming in a few days or so :) Uhh, that linked code looks messy ;)

**Comment by OverSave** [comment]
Okay xD

**Comment by klohner** [comment]
This seems to be working now. Issue closed?
```
$ rtl_433 -f 250k -r rtl_433_tests/tests/auriol/Auriol_HG02832D/g001_433.93M_250k.cu8

time      : @0.285064s
model     : Auriol-HG02832                         id        : 34
channel   : 1            Battery   : 1             Temperature: 15.7 C       Humidity  : 84 %          Button    : 0             Integrity : CRC
```

**Comment by merbanan** [comment]
Decoder implemented so closing ticket.

**Comment by NilsBe** [comment]
Great Work! Received my SDR today and installed rtl_433. Could receive the readings from my Auriol-HG02832 sensor. Thank you all!

**Comment by zuckschwerdt** [comment]
Good to hear! Makes me think, maybe we need a simple feedback form where users can enter their device with a simple works/doesn't work checkbox /counter. A tally of what's used and working vs. needs work.
